llmstack/common/utils/splitter.py

```diff
@@ -216,3 +218,42 @@ def split_text(self, text: str) -> List[str]:
         soup = BeautifulSoup(text, "html.parser")
         html_chunks = self._split_html(soup)
         return self._merge_chunks(html_chunks, separator="")
+
+
+class UnstructuredTextSplitter(TextSplitter):
+    """Interface for splitting unstructured text into structured data."""
+
+    def __init__(self, chunk_size: int = 4000, length_function: Any = len):
+        super().__init__(chunk_size, chunk_overlap=0, length_function=length_function)
+
+    def _split_text(self, text: str) -> List[str]:
+        elements = partition_text(
+            text=text,
+            skip_infer_table_types="[]",  # don't forget to include apostrophe around the square bracket
+        )
+        chunks = chunk_by_title(elements, max_characters=self._chunk_size)
+        return [chunk.text for chunk in chunks]
+
+    def split_text(self, text: str) -> List[str]:
+        chunks = self._split_text(text)
+        return self._merge_chunks(chunks, separator="")
+
+
+class UnstructuredDocumentSplitter(TextSplitter):
+    """Interface for splitting unstructured text into structured data."""
+
+    def __init__(self, file_name: str, chunk_size: int = 4000, length_function: Any = len):
```

Comment:
Looks like unstructured has the ability to provide chunking strategy, can we take that as an argument and default it `by_title` maybe.